cmake/cppfront-config.cmake.in

```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.23)
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+find_dependency(cppfront-exe)
+
+include("${CMAKE_CURRENT_LIST_DIR}/cppfront-targets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/CppfrontHelpers.cmake")
+
+check_required_components(cppfront)
```

Comment:
This package has no _package components_, this check is unnecessary. I think you are confusing them with _install components_.

_Install components_ as defined by `install(... COMPONENT ...)` are used by CPack and `cmake --install ... --component ...` only.

_Package components_ are entirely dealt with in the config script. Here is an [example](https://github.com/friendlyanon/SFML/blob/feature/cmake-rework/cmake/SFMLConfig.cmake.in#L23).

Reply:
No, I'm not. That macro will throw an error if a component is incorrectly specified. Ie. `find_package(cppfront REQUIRED foobardoesnotexist)`.

Reply:
> This macro should be called even if the package doesn't provide any components to make sure users are not specifying components erroneously.
> -- https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html

Reply:
I find it funny that you find prefixing target names hyper-defensive, but this isn't that :)

I guess you're right, this is stricter and doesn't lose anyone anything. "I'm not going to die on this hill" 😀

Reply:
> I find it funny that you find prefixing target names hyper-defensive, but this isn't that :)

@friendlyanon - yes, we all have our own value functions for such things. I won't pretend mine is perfect. :slightly_smiling_face: